### PR TITLE
Release - New GitHub workflow for releasing

### PR DESCRIPTION
**Pull request by araratthehero**
## Description
- Allow running the job on a specified branch instead of main
- Use new Github secrets for publishing the app on maven central
- Make the Github release draft

## Things to do in the future
- Add a step to verify if the branch name starts with e.g. `release/`
- Separate each step into a different job

## Checklist 
- [x] PR is labelled 
- [x] Changes are tested manually

COAND-1013

### CODE REVIEW

**Comment by jreij** [comment]
PR base should not be `main`, the current release process requires `develop` to be up to date with `main` so we cannot merge anything directly to `main` before we are fully ready to remove the old release process

File: .github/workflows/publish_release_manual.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+jobs:
+  
+  publish-release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: main
```
Comment (araratthehero) [inline .github/workflows/publish_release_manual.yml:19]: This needs to change. It should be the branch on which we triggered the action.

**Review by jreij — CHANGES_REQUESTED** [review]
Once we merge the feature branch we will lose the old request action, can we instead keep it and name it something like `old`?

File: .github/workflows/publish_release_old.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          SIGNING_SECRET_KEY_RING_FILE: ${{ secrets.SIGNING_SECRET_KEY_RING_FILE }}
+          SONATYPE_STAGING_PROFILE_ID: ${{ secrets.SONATYPE_STAGING_PROFILE_ID }}
+
+      # Get the version name from a script and save to environment variable.
+      - name: Set PROJECT_VERSION
+        run: |
+          echo "▸ Set run permission."
+          chmod +x scripts/version_name.sh
+          echo "▸ Getting version name"
+          PROJECT_VERSION=$(./scripts/version_name.sh)
```
Comment (araratthehero) [inline .github/workflows/publish_release_old.yml:64]: This need to be updated when [this PR](https://github.com/Adyen/adyen-android/pull/1814) is merged.

Comment (araratthehero) [reply .github/workflows/publish_release_old.yml:64]: Similarly the `publish_release.yml` as well

Comment (araratthehero) [reply .github/workflows/publish_release_old.yml:64]: We do not have to wait for that PR anymore. The changes made [here](https://github.com/Adyen/adyen-android/pull/1848) are enough.

**Review by jreij — APPROVED** [review]

**Pull request merged**